fili-core/src/main/java/com/yahoo/bard/webservice/druid/client/impl/AsyncDruidWebServiceImpl.java

```diff
@@ -97,49 +136,52 @@ public AsyncDruidWebServiceImpl(
             ObjectMapper mapper,
             Supplier<Map<String, String>> headersToAppend
     ) {
-        this(serviceConfig, initializeWebClient(serviceConfig.getTimeout()), mapper, headersToAppend);
+        this(
+                serviceConfig,
+                initializeWebClient(serviceConfig.getTimeout()),
+                mapper,
+                headersToAppend,
+                DEFAULT_JSON_NODE_BUILDER_STRATEGY
+        );
     }
 
     /**
-     * Initialize the client config.
-     *
-     * @param requestTimeout  Timeout to use for the client configuration.
+     * Friendly non-DI constructor useful for manual tests.
      *
-     * @return the set up client
+     * @param serviceConfig  Configuration for the Druid Service
+     * @param mapper  A shared jackson object mapper resource
+     * @param headersToAppend Supplier for map of headers for Druid requests
+     * @param jsonNodeBuilderStrategy A function to build JSON nodes from the response
      */
-    private static AsyncHttpClient initializeWebClient(int requestTimeout) {
-
-        LOG.debug("Druid request timeout: {}ms", requestTimeout);
-
-        // Build the configuration
-        AsyncHttpClientConfig config = new DefaultAsyncHttpClientConfig.Builder()
-                .setReadTimeout(requestTimeout)
-                .setRequestTimeout(requestTimeout)
-                .setConnectTimeout(requestTimeout)
-                .setConnectionTtl(requestTimeout)
-                .setPooledConnectionIdleTimeout(requestTimeout)
-                .setFollowRedirect(true)
-                .build();
-
-        return new DefaultAsyncHttpClient(config);
+    public AsyncDruidWebServiceImpl(
+            DruidServiceConfig serviceConfig,
+            ObjectMapper mapper,
+            Supplier<Map<String, String>> headersToAppend,
+            Function<Response, JsonNode> jsonNodeBuilderStrategy
+    ) {
+        this(
+                serviceConfig,
+                initializeWebClient(serviceConfig.getTimeout()),
+                mapper,
+                headersToAppend,
+                jsonNodeBuilderStrategy
+        );
     }
-
     /**
      * IOC constructor.
```

Comment:
JavaDoc should call out what strategy is used as the default, since this constructor doesn't take a strategy.